Unable to build vmprof on Windows with Python 3.5.2

Hello people.

Linkage failed when building vmprof on Windows with Python 3.5.2 x86-64 and Visual Studio Community 2015 with Python support.

Here the log:

```
C:\Users\Marc>pip install vmprof
Collecting vmprof
  Using cached vmprof-0.2.8.tar.gz
Requirement already satisfied (use --upgrade to upgrade): requests in c:\python35\lib\site-packages (from vmprof)
Installing collected packages: vmprof
  Running setup.py install for vmprof ... error
    Complete output from command c:\python35\python.exe -u -c "import setuptools, tokenize;__file__='D:\\Temp\\Users\\Marc\\pip-build-d2p21sz2\\vmprof\\setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record D:\Temp\Users\Marc\pip-dtex470n-record\install-record.txt --single-version-externally-managed --compile:
    running install
    running build
    running build_py
    creating build
    creating build\lib.win-amd64-3.5
    creating build\lib.win-amd64-3.5\tests
    copying tests\cpuburn.py -> build\lib.win-amd64-3.5\tests
    copying tests\test_config.py -> build\lib.win-amd64-3.5\tests
    copying tests\test_reader.py -> build\lib.win-amd64-3.5\tests
    copying tests\test_run.py -> build\lib.win-amd64-3.5\tests
    copying tests\test_stats.py -> build\lib.win-amd64-3.5\tests
    copying tests\__init__.py -> build\lib.win-amd64-3.5\tests
    creating build\lib.win-amd64-3.5\vmprof
    copying vmprof\binary.py -> build\lib.win-amd64-3.5\vmprof
    copying vmprof\cli.py -> build\lib.win-amd64-3.5\vmprof
    copying vmprof\profiler.py -> build\lib.win-amd64-3.5\vmprof
    copying vmprof\reader.py -> build\lib.win-amd64-3.5\vmprof
    copying vmprof\show.py -> build\lib.win-amd64-3.5\vmprof
    copying vmprof\stats.py -> build\lib.win-amd64-3.5\vmprof
    copying vmprof\upload.py -> build\lib.win-amd64-3.5\vmprof
    copying vmprof\vmprofdemo.py -> build\lib.win-amd64-3.5\vmprof
    copying vmprof\__init__.py -> build\lib.win-amd64-3.5\vmprof
    copying vmprof\__main__.py -> build\lib.win-amd64-3.5\vmprof
    creating build\lib.win-amd64-3.5\vmprof\log
    copying vmprof\log\constants.py -> build\lib.win-amd64-3.5\vmprof\log
    copying vmprof\log\marks.py -> build\lib.win-amd64-3.5\vmprof\log
    copying vmprof\log\merge_point.py -> build\lib.win-amd64-3.5\vmprof\log
    copying vmprof\log\objects.py -> build\lib.win-amd64-3.5\vmprof\log
    copying vmprof\log\parser.py -> build\lib.win-amd64-3.5\vmprof\log
    copying vmprof\log\__init__.py -> build\lib.win-amd64-3.5\vmprof\log
    running egg_info
    writing requirements to vmprof.egg-info\requires.txt
    writing top-level names to vmprof.egg-info\top_level.txt
    writing vmprof.egg-info\PKG-INFO
    writing entry points to vmprof.egg-info\entry_points.txt
    writing dependency_links to vmprof.egg-info\dependency_links.txt
    warning: manifest_maker: standard file '-c' not found

    reading manifest file 'vmprof.egg-info\SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    writing manifest file 'vmprof.egg-info\SOURCES.txt'
    running build_ext
    building '_vmprof' extension
    creating build\temp.win-amd64-3.5
    creating build\temp.win-amd64-3.5\Release
    creating build\temp.win-amd64-3.5\Release\src
    C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\amd64\cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MD -Ic:\python35\include -Ic:\python35\include "-IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\INCLUDE" "-IC:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\ATLMFC\INCLUDE" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.10586.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\include\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.10586.0\shared" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.10586.0\um" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.10586.0\winrt" /Tcsrc/_vmprof.c /Fobuild\temp.win-amd64-3.5\Release\src/_vmprof.obj
    _vmprof.c
    d:\temp\users\marc\pip-build-d2p21sz2\vmprof\src\vmprof_common.h(67): warning C4311: 'cast de type' : troncation de pointeur de 'PyCodeObject *' à 'unsigned long'
    d:\temp\users\marc\pip-build-d2p21sz2\vmprof\src\vmprof_common.h(67): warning C4312: 'cast de type' : la conversion de 'unsigned long' en 'void *' d'une taille supérieure
    d:\temp\users\marc\pip-build-d2p21sz2\vmprof\src\vmprof_common.h(96): warning C4267: '=' : conversion de 'size_t' en 'char', perte possible de données
    d:\temp\users\marc\pip-build-d2p21sz2\vmprof\src\vmprof_main_win32.h(31): warning C4267: 'fonction' : conversion de 'size_t' en 'unsigned int', perte possible de données
    d:\temp\users\marc\pip-build-d2p21sz2\vmprof\src\vmprof_main_win32.h(48): warning C4267: 'initialisation' : conversion de 'size_t' en 'int', perte possible de données
    d:\temp\users\marc\pip-build-d2p21sz2\vmprof\src\vmprof_main_win32.h(72): warning C4312: 'cast de type' : la conversion de 'DWORD' en 'void *' d'une taille supérieure
    src/_vmprof.c(42): warning C4311: 'cast de type' : troncation de pointeur de 'PyCodeObject *' à 'unsigned long'
    src/_vmprof.c(42): warning C4312: 'cast de type' : la conversion de 'unsigned long' en 'void *' d'une taille supérieure
    src/_vmprof.c(69): warning C4311: 'cast de type' : troncation de pointeur de 'PyCodeObject *' à 'unsigned long'
    C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\amd64\link.exe /nologo /INCREMENTAL:NO /LTCG /DLL /MANIFEST:EMBED,ID=2 /MANIFESTUAC:NO /LIBPATH:c:\python35\libs /LIBPATH:c:\python35\PCbuild\amd64 "/LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\LIB\amd64" "/LIBPATH:C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\ATLMFC\LIB\amd64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.10586.0\ucrt\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\NETFXSDK\4.6.1\lib\um\x64" "/LIBPATH:C:\Program Files (x86)\Windows Kits\10\lib\10.0.10586.0\um\x64" /EXPORT:PyInit__vmprof build\temp.win-amd64-3.5\Release\src/_vmprof.obj /OUT:build\lib.win-amd64-3.5\_vmprof.cp35-win_amd64.pyd /IMPLIB:build\temp.win-amd64-3.5\Release\src\_vmprof.cp35-win_amd64.lib
    _vmprof.obj : warning LNK4197: exportation 'PyInit__vmprof' spÚcifiÚe Ó plusieurs reprisesá; premiÞre spÚcification utilisÚe
       CrÚation de la bibliothÞque build\temp.win-amd64-3.5\Release\src\_vmprof.cp35-win_amd64.lib et de l'objet build\temp.win-amd64-3.5\Release\src\_vmprof.cp35-win_amd64.exp
    _vmprof.obj : error LNK2001: symbole externe non rÚsolu _PyThreadState_Current
    build\lib.win-amd64-3.5\_vmprof.cp35-win_amd64.pyd : fatal error LNK1120: 1 externes non rÚsolus
    error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio 14.0\\VC\\BIN\\amd64\\link.exe' failed with exit status 1120

    ----------------------------------------
Command "c:\python35\python.exe -u -c "import setuptools, tokenize;__file__='D:\\Temp\\Users\\Marc\\pip-build-d2p21sz2\\vmprof\\setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" install --record D:\Temp\Users\Marc\pip-dtex470n-record\install-record.txt --single-version-externally-managed --compile" failed with error code 1 in D:\Temp\Users\Marc\pip-build-d2p21sz2\vmprof\
```

Thank you. ;)

--- a/src/vmprof_common.h
+++ b/src/vmprof_common.h
@@ -136,9 +136,9 @@ PyThreadState * get_current_thread_state(void)
 {
 #if PY_MAJOR_VERSION < 3
     return _PyThreadState_Current;
-#elif PY_VERSION_HEX < 0x03050000
+#elif PY_VERSION_HEX < 0x03050200
     return (PyThreadState*) _Py_atomic_load_relaxed(&_PyThreadState_Current);
 #else
-    return PyThreadState_GET();
+    return _PyThreadState_UncheckedGet();
 #endif
 }

--- a/src/vmprof_common.h
+++ b/src/vmprof_common.h
@@ -132,19 +132,13 @@ static int opened_profile(char *interp_name, int memory, int lines)
     return _write_all((char*)&header, 5 * sizeof(long) + 5 + namelen);
 }
 
-// for whatever reason python-dev decided to hide that one
-#if PY_MAJOR_VERSION >= 3 && !defined(_Py_atomic_load_relaxed)
-                                 /* this was abruptly un-defined in 3.5.1 */
-    extern void *volatile _PyThreadState_Current;
-       /* XXX simple volatile access is assumed atomic */
-#  define _Py_atomic_load_relaxed(pp)  (*(pp))
-#endif
- 
-PyThreadState* get_current_thread_state(void)
+PyThreadState * get_current_thread_state(void)
 {
-#if PY_MAJOR_VERSION >= 3
-    return (PyThreadState*)_Py_atomic_load_relaxed(&_PyThreadState_Current);
-#else
+#if PY_MAJOR_VERSION < 3
     return _PyThreadState_Current;
+#elif PY_VERSION_HEX < 0x03050000
+    return (PyThreadState*) _Py_atomic_load_relaxed(&_PyThreadState_Current);
+#else
+    return PyThreadState_GET();
 #endif
 }

